store/RedisStore.js

```diff
@@ -5,46 +5,70 @@ function RedisStore (redis, timeWindow, continueExceeding, key) {
   this.timeWindow = timeWindow
   this.continueExceeding = continueExceeding
   this.key = key
+
+  if (this.redis.options.enableAutoPipelining) {
+    RedisStore.prototype.incr = RedisStore.prototype.incrAutoPipeline
+  } else {
+    RedisStore.prototype.incr = RedisStore.prototype.incrNormal
+  }
 }
 
-RedisStore.prototype.incr = function (ip, cb, max) {
+RedisStore.prototype.incrAutoPipeline = async function (ip, cb, max) {
   const key = this.key + ip
+  const [current, ttl] = await Promise.all([this.redis.incr(key), this.redis.pttl(key)])
+
+  if (ttl === -1) {
+    // Item just got created
+    this.redis.pexpire(key, this.timeWindow).catch()
+    cb(null, { current: 1, ttl: this.timeWindow })
```

Comment:
I think we need to add these as callback to pexpire to avoid test issues

Reply:
Yeah I think that’s an improvement, just wanted to keep the behavior the same - it was like this before even with callbacks

Reply:
What do you think about making the functions async? Without making them async, I can’t group them together without .pipeline

Reply:
that would be a semver-major.

Reply:
+1 for me

Reply:
I really would prefer a customCommand which runs the db related stuff on redis over more fiddling in the business code.